app/code/core/Mage/Cron/Model/Observer.php

```diff
@@ -220,7 +220,7 @@ public function cleanup()
 
         $now = time();
         foreach ($history->getIterator() as $record) {
-            if (strtotime($record->getExecutedAt()) < $now - $historyLifetimes[$record->getStatus()]) {
+            if (strtotime($record->getExecutedAt() ?? '') < $now - $historyLifetimes[$record->getStatus()]) {
```

Comment:
I am not sure if this is better, but the intent is clearer:

```suggestion
            if (empty($record->getExecutedAt())
                || (strtotime($record->getExecutedAt()) < $now - $historyLifetimes[$record->getStatus()])
            ) {
```

Note that `strtotime('')` is `false`. The comparison is odd `if (false < $someValue)`.

Reply:
correct at always, I was sure strtotime('') would give the current timestamp...